test/unit/org/apache/cassandra/cql3/DurationTest.java

```diff
@@ -174,7 +175,211 @@ private long toMillis(String timeAsString)
         return DateUtils.truncate(date, Calendar.SECOND).getTime();
     }
 
-    public void assertInvalidDuration(String duration, String expectedErrorMessage)
+    @Test
+    public void testInvalidFloorTimestamp()
+    {
+        try
+        {
+            floorTimestamp("2016-09-27T16:12:00", "2h", "2017-09-01T00:00:00");
+            Assert.fail();
+        }
+        catch (InvalidRequestException e)
+        {
+            assertEquals("The floor function starting time is greater than the provided time", e.getMessage());
+        }
+
+        try
+        {
+            floorTimestamp("2016-09-27T16:12:00", "-2h", "2016-09-27T00:00:00");
+            Assert.fail();
+        }
+        catch (InvalidRequestException e)
+        {
+            assertEquals("Negative durations are not supported by the floor function", e.getMessage());
+        }
+    }
+
+    @Test
+    public void testFloorTimestamp()
+    {
+        //~ Test with a duration in hours //////////////////////////////////////////////
```

Comment:
Seems like we could have a separate test method for each of these `//~` headers, wdyt?